avocado/utils/network/interfaces.py

```diff
@@ -144,14 +144,32 @@ def bring_up(self):
         except Exception as ex:
             raise NWException("Failed to bring up: %s" % ex)
 
+    def get_operational_link_state(self):
+        """Get Operational link state.
+
+        :return: Operational link state.Or it will raise a NWException if is
+                 unable to get the operational link state.
+        """
+        try:
+            return self._get_interface_details().get('flags')[3]
+        except (NWException, IndexError):
+            raise NWException("Could not get operational link state.")
+
     def is_link_up(self):
         """Check if the interface is up or not.
 
-        :return: True or False. True if the current state is UP, otherwise
-                 will return False.
+        :return: True or False. True if admin state is UP and operational state
+                 is LOWER_UP or if admin state is unknown and operational state
+                 is LOWER_UP, otherwise will return False.
         """
-        if self.get_link_state() == 'up':
-            return True
+        link_state = self.get_admin_link_state()
+        oper_state = self.get_operational_link_state()
+        if link_state == 'up':
+            if oper_state == 'LOWER_UP':
+                return True
+        elif link_state == 'unknown':
+            if oper_state == 'LOWER_UP':
+                return True
```

Comment:
This is the way to go. But IMO we should split the methods, so this method should be very small:

```python
def is_link_up(self):
   return self.is_link_admin_up() and self.is_link_operational_up()
```

Also if we are renaming the `is_link_up()` method this should be deprecated first, by adding a warning message AND adding the new method. We should remove this method, only after a few minor-releases.

Reply:
ok